src/cloud.rs

```diff
@@ -611,6 +613,25 @@ impl Cloud {
         self.find_subnets().all()
     }
 
+    /// Prepare a new object for creation.
+    ///
+    /// This call returns a `NewObject` object, which is a builder
+    /// to create object in object storage.
+    #[cfg(feature = "network")]
+    pub fn new_object<C, O, B>(&self, container: C, object: O, body: B) -> NewObject<B>
+    where
+        C: Into<String>,
+        O: Into<String>,
+        B: io::Read + Send + 'static,
```

Comment:
nit: you could avoid `'static` probably, but it's fine this way (and will probably change with async/await anyway).

Reply:
For the `'static` if I don't put it I have a compilation error.

I don't know yet about the async/await, as I told you I just begin in Rust ;)

Reply:
Yeah, it's not as simple as just removing `'static` :) To put it simply, static means that the received object won't have references to anything outside of it (normal files, for example, fulfil this).

We could use rust lifetime syntax to express that `B` has to outlive the resulting `NewObject`. I'm not sure if it's useful. Ignore it for now.

By async/await I mean https://rust-lang.github.io/async-book/01_getting_started/04_async_await_primer.html. I'd like to migrate rust-openstack to it in the next major version (coming in weeks.. or months...).